pkgs/applications/misc/dstask/default.nix

```
@@ -0,0 +1,41 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "dstask";
+  version = "0.18";
+  rev = "d687a0f9aec634a808cfa0df4aecb06ab4f57732";
+
+  src = fetchFromGitHub {
+    owner = "naggie";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "16z5zlfj955pzsj0l58835slvpchdaq2vbyx2fjzi6y9xn1z2nd1";
+  };
+
+  # Set vendorSha256 to null because dstask vendors its dependencies (meaning
+  # that third party dependencies are stored in the repository).
+  #
+  # Ref <https://github.com/NixOS/nixpkgs/pull/87383#issuecomment-633204382>
+  # and <https://github.com/NixOS/nixpkgs/blob/d4226e3a4b5fcf988027147164e86665d382bbfa/pkgs/development/go-modules/generic/default.nix#L18>
+  vendorSha256 = null;
+
+  # The ldflags reduce the executable size by stripping some debug stuff.
+  # The other variables are set so that the output of dstask version shows the
+  # git ref and the release version from github.
+  # Ref <https://github.com/NixOS/nixpkgs/pull/87383#discussion_r432097657>
+  buildFlagsArray = [ ''
+    -ldflags=-w -s
+    -X "github.com/naggie/dstask.VERSION=v${version}"
+    -X "github.com/naggie/dstask.GIT_COMMIT=${rev}"
```

How about this:

```suggestion
    -X "github.com/naggie/dstask.VERSION=${version}"
    -X "github.com/naggie/dstask.GIT_COMMIT=v${version}"
```

I don't see why there should be a `v` prefix for the version. At the same time, `v${version}` is the name of the tag and therefore a valid specifier of a git commit.

---

Seems like a good change as it would make the output of dstask version consistent with the official binary, which currently outputs:

```
Version: v0.18
Git commit: d687a0f9aec634a808cfa0df4aecb06ab4f57732
Build date: Wed 13 May 18:34:38 BST 2020

```

---

Thank you both again! I've updated the PR.

---

@naggie it sound like you may be reading the diff the wrong way around. Previously it would have output

```
Version: v0.18
Git commit: d687a0f9aec634a808cfa0df4aecb06ab4f57732
```

but now it should be

```
Version: 0.18
Git commit: v0.18
```

The reasoning behind this is that we would like to avoid having to manually set the fully-qualified git revision of the release in our expression. That would be cumbersome and our automated tooling wouldn't update that. At the same time, since the version is tagged `v0.18` is also a valid reference to the git revision.

We've dropped the `v` prefix from `Version` since `0.18` is the version number while `v0.18` is the name of the tag.

Is that fine by you?